### PR TITLE
feat: Check if run has already happened

### PR DESCRIPTION
**Pull request by JohannesWesch**
# Description
No description.

## Before Merging
 - [x] Review the code changes
    - Unused print / comments / TODOs
    - Missing docstrings for functions that should have them
    - Consistent variable names
    - ...
 - [x] Update `changelog.md` if necessary
 - [x] Commit messages should contain a semantic [label](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) and the ticket number
   - Consider squashing if this is not the case

### CODE REVIEW

File: src/documentation/parameter_optimization.ipynb [hunk 1]
```diff
@@ -299,11 +299,61 @@
...
+    "# Let's print the number of run overviews. Because we have not recomputed the 12 (3 models times 4 examples) runs we now expect 16 runs in total.\n",
+    "# Without the `recompute_if_metadata_changed` flag we would get 28 runs!\n",
+    "print(len(list(run_repository.run_overviews())))"
+   ]
   }
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": ".venv",
```
Comment (SebastianNiehusAA) [inline src/documentation/parameter_optimization.ipynb:356]:
```suggestion
   "display_name": "Python 3,
```

File: src/documentation/parameter_optimization.ipynb [hunk 2]
```diff
@@ -299,11 +299,61 @@
...
+    "# Let's print the number of run overviews. Because we have not recomputed the 12 (3 models times 4 examples) runs we now expect 16 runs in total.\n",
+    "# Without the `recompute_if_metadata_changed` flag we would get 28 runs!\n",
+    "print(len(list(run_repository.run_overviews())))"
+   ]
   }
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 3,
```
Comment (SebastianNiehusAA) [inline src/documentation/parameter_optimization.ipynb:356]:
```suggestion
   "display_name": "Python 3",
```

File: src/documentation/how_tos/how_to_implement_incremental_evaluation.ipynb
Comment (SebastianNiehusAA) [inline src/documentation/how_tos/how_to_implement_incremental_evaluation.ipynb:1]: Changes in this file are not directly connected to the new functionality, but we fixed it while we noticed it

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,8 @@
 ## Unreleased
 
 ### Features
-...
+- Add Catalan and Polish support to `detectLanguage`.
```
Comment (NiklasKoehneckeAA) [inline CHANGELOG.md:5]: detectLanguage seems to have weird casing, are you sure that its called like this?

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,7 +2,8 @@
...
+- Add utility function `run_is_already_computed` to `Runner` to check if a run with the given metadata has already been computed.
```
Comment (NiklasKoehneckeAA) [inline CHANGELOG.md:6]: you don't mention the `retrieve_previous_runs`. I would also mention the changes to the notebook so people know how to use it

**Pull request merged**